### PR TITLE
ref(platform-picker): Add language category icon

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- A new language icon can now be shown in the bottom right on the platform picker Using the `withLanguageIcon` prop.

 - Sorts the platforms in the platform list so the language shows up first in the list.

 - Adjusts the platform picker nav to allow more categories to fit

 This is a follow up to https://github.com/getsentry/sentry/pull/20723

image

### CODE REVIEW

File: src/sentry/static/sentry/app/components/platformIcon.tsx [hunk 1]
```diff
@@ -92,6 +107,20 @@ const PlatformIcon = ({platform, size, ...props}: Props) => {
     size === 'lg' ? 'svg_80x80' : 'svg'
   }/${icon}.svg`);
 
+  const language = platform.split('-')[0];
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/platformIcon.tsx:110]: Personal style preference is to destructure

**Review by billyvg — COMMENTED** [review]
This looks great 😍

From the screenshot it doesn't look like sorting is correct, e.g. Swift before Objective-c - also, the java ones (Java and java.util.logging).

**Comment by EvanPurkhiser** [comment]
> From the screenshot it doesn't look like sorting is correct, e.g. Swift before Objective-c - also, the java ones (Java and java.util.logging).

This is sort of an artifact of the nature of these. `swift` is actually it's own language and

**Comment by EvanPurkhiser** [comment]
> From the screenshot it doesn't look like sorting is correct, e.g. Swift before Objective-c - also, the java ones (Java and java.util.logging).

Hmm, I'm actually not sure why Swift is showing up twice in the first place 🤔

I wouldn't mind merging this first and coming back and fixing the sorting a bit later, since it wasn't sorted at all before afaict

**Review by billyvg — APPROVED** [review]

**Pull request merged**